src/constants.ts

```diff
@@ -20,3 +24,22 @@ export const EXPLORER_SUBSCRIPTION_TERMINATION =
   'ExplorerSubscriptionTermination';
 export const IFRAME_DOM_ID = (uniqueId: number) =>
   `apollo-embedded-explorer-${uniqueId}`;
+
+// Message types for authentication
+export const EXPLORER_LISTENING_FOR_HANDSHAKE = 'ExplorerListeningForHandshake';
+export const HANDSHAKE_RESPONSE = 'HandshakeResponse';
+
+export type OutgoingEmbedMessage =
+  | {
+      name: 'SchemaResponse';
+      schema: IntrospectionQuery | string | undefined;
+    }
+  | {
+      name: 'HandshakeResponse';
+      graphRef: string | undefined;
+    }
+  | {
+      name: 'ExplorerResponse';
+      operationId: string;
+      response: ApolloQueryResult<JSONValue>;
```

Comment:
Ah sorry this is wrong in studio-ui and I have an open ticket to fix it. 

Can make this type `response: Response` and not have apollo-client as a dependency & delete the `types.ts` file. I will fix this in studio-ui today!

Reply:
actually i lied this isn't a Response object. This is probably going to be an issue in the future when we want to make breaking changes like switching this to a Response object 😢 . 

Lets make this type 

type Error = {
message: string;
}

{
data?: JSONValue
error?: Error
errors?: [Error]
}